var/spack/repos/builtin/packages/hipsycl/package.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+from os import path
+from llnl.util import filesystem
+
+
+class Hipsycl(CMakePackage):
+    """hipSYCL is an implementation of the SYCL standard programming model
+    over NVIDIA CUDA/AMD HIP"""
+
+    homepage = "https://github.com/illuhad/hipSYCL"
+    url = "https://github.com/illuhad/hipSYCL/archive/v0.8.0.tar.gz"
+    git = "https://github.com/illuhad/hipSYCL.git"
+
+    maintainers = ["nazavode"]
+
+    provides("sycl")
+
+    version("master", branch="master")
+    version(
+        "0.8.0",
+        sha256="4ec5218103d1b38efae9a51ca752b9b44bbd02dada78c05e20e00c9c25e9ea19",
+    )
+
+    variant("cuda", default=False, description="Enable CUDA backend for SYCL kernels")
+
+    depends_on("boost +filesystem")
+    depends_on("python@3:")
+    depends_on("llvm@8: +clang", when="~cuda")
+    # LLVM debug builds don't work with hipSYCL CUDA backend:
+    # https://github.com/illuhad/hipSYCL/blob/master/doc/install-cuda.md
+    depends_on("llvm@9: +clang build_type=Release", when="+cuda")
+    # hipSYCL requires cuda@9:
+    # LLVM PTX backend requires cuda7:10.1.9999 (https://tinyurl.com/v82k5qq)
+    depends_on("cuda@9:10.1.9999", when="+cuda")
+
+    depends_on("cmake@3.5:", type="build")
+
+    def cmake_args(self):
+        spec = self.spec
+        args = [
+            "-DLLVM_DIR:String={0}".format(
+                path.dirname(
+                    filesystem.find(spec["llvm"].prefix, "LLVMExports.cmake"))
+            ),
+            "-DCLANG_EXECUTABLE_PATH:String={0}".format(
+                path.join(spec["llvm"].bin, "clang")
+            ),
+            "-DCLANG_INCLUDE_PATH:String={0}".format(
+                path.join(spec["llvm"].include, "clang")
```

Comment:
I'm not very familiar with spack, so I don't really know what will come out here. Just to be sure, this needs to be expanded to the *internal* clang include directory which is usually at a location like `$PREFIX/llvm/lib/clang/$CLANG_VERSION/include`. It has to be the directory with the clang CUDA wrappers inside, not the directory with the libclang header files.
Note: On CUDA, this setting is usually not required and it might just work even if it's set to the wrong location. But it's very important for the ROCm backend.

Reply:
It should be fixed, thanks for the hint. That explains why with CUDA I never had a single error even if I was including the wrong directory :)